### ISSUE TITLE
`struct_field_names` not triggered when 2 or less fields

### ISSUE THREAD

**Issue by alexis-langlet** [rust-lang/rust-clippy#12089]
### Summary

As mentioned in #12035, `struct_field_names` lint is only triggered when struct contains more than 3 fields. 
It should be triggered no matter the fields' number.

### Lint Name

struct_field_names

### Reproducer

I tried this code:

```rust
#![warn(clippy::pedantic)]
struct A {
    a_group: String,
    a_location: String,
}
```
struct_field_names
I expected to see this happen: `warning: field name starts with the struct's name`

Instead, this happened: no warning about it

### Version

```text
rustc 1.76.0-nightly (87e1447aa 2023-11-30)
binary: rustc
commit-hash: 87e1447aadaa2899ff6ccabe1fa669eb50fb60a1
commit-date: 2023-11-30
host: x86_64-unknown-linux-gnu
release: 1.76.0-nightly
LLVM version: 17.0.5
```

**Comment by y21** [comment]
FWIW, the reasoning for this was consistency with the enum_variant_names lint, which also only lints when there's a minimum number of variants: #1138.
That threshold can also be changed through a configuration option: https://doc.rust-lang.org/nightly/clippy/lint_configuration.html#struct-field-name-threshold

**Comment by alexis-langlet** [comment]
@y21 It looks like this one does not depends on a number or variants but only on a number of fields
e.g.
This triggers:
```rust
#![warn(clippy::pedantic)]
struct A {
    a_id: i32,
    a_group: String,
    a_location: String,
}
but this also triggers: 
```rust
#![warn(clippy::pedantic)]
struct A {
    id: i32,
    group: String,
    a_location: String,
}
```

Is this what the threshold is suppose to do?

**Comment by alexis-langlet** [comment]
Ok, my bad.
I read it again and it looks like the threshold is actually about the number of fields in the struct

Thank you very much for the help and sorry for the inconvenience :sweat_smile:

**Comment by y21** [comment]
No worries. I do think that we should at least document some of this in the lint description (like, mention that the threshold exists and that it is configurable, and why it exists). It's not very discoverable and I don't blame anyone for being confused by this 🙃
